src/auth/auth.repository.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from "@nestjs/common";
+
+import { EntityManager } from "@mikro-orm/core";
+
+import * as bcrypt from "bcrypt";
+
+import { Student } from "@/common/entities/students.entity";
+import { User } from "@/common/entities/users.entity";
+import { EMedium } from "@/common/enums/students.enums";
+import { EUserType } from "@/common/enums/users.enums";
+
+import { RegisterStudentDto } from "./auth.dtos";
+
+@Injectable()
+export class AuthRepository {
+  constructor(private readonly em: EntityManager) {}
+
+  async createStudent(registerStudentDto: RegisterStudentDto): Promise<User> {
+    const saltRounds = 10;
+    const hashedPassword = await bcrypt.hash(registerStudentDto.password, saltRounds);
+
+    const user = this.em.create(User, {
+      ...registerStudentDto,
+      password: hashedPassword,
+      userType: EUserType.STUDENT,
+    });
+
+    const medium =
+      registerStudentDto.medium === undefined ? EMedium.NONE : registerStudentDto.medium;
+
+    const student = this.em.create(Student, {
+      educationLevel: registerStudentDto.educationLevel,
+      medium: medium,
+      class: registerStudentDto.class,
+      degree: registerStudentDto.degree,
+      degreeName: registerStudentDto.degreeName,
+      semester: registerStudentDto.semester,
+      user: user,
+    });
+
+    await this.em.persistAndFlush([user, student]);
```

Comment:
This should be a transaction. 
DB should rollback if one or the other db action fails. Look into [transactions from mikroORM docs](https://mikro-orm.io/docs/transactions)